L1Trigger/L1TGlobal/plugins/L1TGlobalProducer.cc

```diff
@@ -103,6 +105,7 @@ L1TGlobalProducer::L1TGlobalProducer(const edm::ParameterSet& parSet)
       m_tauInputTag(parSet.getParameter<edm::InputTag>("TauInputTag")),
       m_jetInputTag(parSet.getParameter<edm::InputTag>("JetInputTag")),
       m_sumInputTag(parSet.getParameter<edm::InputTag>("EtSumInputTag")),
+      m_zdcEtSumInputTag(parSet.getParameter<edm::InputTag>("ZdcEtSumInputTag")),
```

Comment:
What should be the value of this new InputTag in the instance of the L1T emulator used inside the HLT menu [1] ?

As far as I understand, in principle it should be a product coming from the L1T unpacker, e.g. something like `"hltGtStage2Digis:ZdcEtSum"`, but I'm not seeing where this is produced. Is there a L1T unpacker for ZDC ? (sorry if the question is naive)

[1]
```
process.hltGtStage2ObjectMap = cms.EDProducer( "L1TGlobalProducer",
    MuonInputTag = cms.InputTag( 'hltGtStage2Digis','Muon' ),
    MuonShowerInputTag = cms.InputTag( 'hltGtStage2Digis','MuonShower' ),
    EGammaInputTag = cms.InputTag( 'hltGtStage2Digis','EGamma' ),
    TauInputTag = cms.InputTag( 'hltGtStage2Digis','Tau' ),
    JetInputTag = cms.InputTag( 'hltGtStage2Digis','Jet' ),
    EtSumInputTag = cms.InputTag( 'hltGtStage2Digis','EtSum' ),
    ExtInputTag = cms.InputTag( "hltGtStage2Digis" ),
    AlgoBlkInputTag = cms.InputTag( "hltGtStage2Digis" ),
    GetPrescaleColumnFromData = cms.bool( False ),
    AlgorithmTriggersUnprescaled = cms.bool( True ),
    RequireMenuToMatchAlgoBlkInput = cms.bool( True ),
    AlgorithmTriggersUnmasked = cms.bool( True ),
    useMuonShowers = cms.bool( True ),
    resetPSCountersEachLumiSec = cms.bool( True ),
    semiRandomInitialPSCounters = cms.bool( False ),
    ProduceL1GtDaqRecord = cms.bool( True ),
    ProduceL1GtObjectMapRecord = cms.bool( True ),
    EmulateBxInEvent = cms.int32( 1 ),
    L1DataBxInEvent = cms.int32( 5 ),
    AlternativeNrBxBoardDaq = cms.uint32( 0 ),
    BstLengthBytes = cms.int32( -1 ),
    PrescaleSet = cms.uint32( 1 ),
    Verbosity = cms.untracked.int32( 0 ),
    PrintL1Menu = cms.untracked.bool( False ),
    TriggerMenuLuminosity = cms.string( "startup" )
)
```

Reply:
```simCaloStage2Digi``` looks like simulation - what about real data?

Reply:
I believe that sim refers to the fact that we are simulating the trigger decision, but the inputs are real data - I've put a snippet from an edmConfigDump below of a python file I run over 2018 PbPb data to test the code. Please correct me if I've misunderstood, however

`
process.simCaloStage2Digis = cms.EDProducer("L1TStage2Layer2Producer",
    firmware = cms.int32(1),
    towerToken = cms.InputTag("simCaloStage2Layer1Digis"),
    useStaticConfig = cms.bool(False)
)

process.simCaloStage2Layer1Digis = cms.EDProducer("L1TCaloLayer1",
    ecalToken = cms.InputTag("ecalDigis","EcalTriggerPrimitives"),
    firmwareVersion = cms.int32(3),
    hcalToken = cms.InputTag("hcalDigis"),
    unpackEcalMask = cms.bool(False),
    unpackHcalMask = cms.bool(False),
    useCalib = cms.bool(True),
    useECALLUT = cms.bool(True),
    useHCALFBLUT = cms.bool(False),
    useHCALLUT = cms.bool(True),
    useHFLUT = cms.bool(True),
    useLSB = cms.bool(True),
    verbose = cms.untracked.bool(False)
)
`

Reply:
Hi Andrea, 
my understanding of the L1 Trigger emulation is that unpackers for objects are running only if there is something to unpack (= ZDC FED present in the run). I let L1T experts comment more though.

On my end, I quickly tried to run the current emulator on a data file from 2022A (that doesn't contain ZDC) but I have an error at earlier emulation stages. Not sure if that's relevant.  
```
----- Begin Fatal Exception 24-Aug-2023 22:00:55 CEST-----------------------
An exception of category 'Conditions mismatch' occurred while
   [0] Processing  Event run: 362219 lumi: 1 event: 14289 stream: 0
   [1] Running path 'L1TReEmulPath'
   [2] Prefetching for module HcalTrigPrimDigiProducer/'simHcalTriggerPrimitiveDigis'
   [3] Calling method for EventSetup module HcalTPGCoderULUT/''
Exception Message:
Requested conditions of type HcalChannelQuality for cell (0x43280401) (HB 1,1,2) got conditions for cell (0x0) 
----- End Fatal Exception -------------------------------------------------
```

Any suggestion about how we can test this effect (and prevent issues) more in detail?

Reply:
@cms-sw/l1-l2 can you suggest how to try running this version of the uGT emulator on older data that does not contain the ZDC ?